docs/resources/samples/copy-tables-combine.md

```diff
@@ -0,0 +1,93 @@
+---
+title: 'Copy Excel tables into new master table'
+description: 'Learn how to use Office Scripts to combine data from multiple Excel tables into a single table.'
+ms.date: 03/04/2021
+localization_priority: Normal
+---
+
+# Copy Excel tables into new master table
+
+Shows how to combine data that resides in Excel tables into a new master table with all rows included. It creates a new worksheet and combines all rows into a single table.
+
+There are two variations of this script:
+
+1. The first script combines all tables in the Excel file. It assumes that all tables in the Excel files have same structure.
+1. The second script selectively gets tables within a set of worksheets. It assumes that all the tables in those select worksheets have the same structure.
+
+## Office Scripts
+
+This solution showcases two Office Scripts:
+
+1. [Copy all tables in an Excel file to a new master table](#copy-all-tables-in-excel-to-a-new-master-table)
+1. [Copy tables from select worksheets into a new master table](#copy-tables-from-select-worksheets-into-a-new-master-table)
+
+### Copy all tables in Excel to a new master table
+
+Download the sample file <a href="tables-copy.xlsx">tables-copy.xlsx</a> and use it with the following script to try it out yourself!
+
+```ts
+function main(workbook: ExcelScript.Workbook) {
+    workbook.getWorksheet('Combined')?.delete();
+    const newSheet = workbook.addWorksheet('Combined');
+    
+    const tables = workbook.getTables();    
+    const headerValues = tables[0].getHeaderRowRange().getTexts();
+    console.log(headerValues);
+    const targetRange = updateRange(newSheet, headerValues);
+    const combinedTable = newSheet.addTable(targetRange.getAddress(), true);
+    for (let table of tables) {      
+      let dataValues = table.getRangeBetweenHeaderAndTotal().getTexts();
+      let rowCount = table.getRowCount();
+      if (rowCount > 0) {
+        combinedTable.addRows(-1, dataValues);
+      }
+    }
+}
+
+function updateRange(sheet: ExcelScript.Worksheet, data: string[][]): ExcelScript.Range {
```

Comment:
Meta question for you:
Is this single-use function separation helping readability? I've seen this pattern a lot in these samples. The alternative would be to just have sequential code.